### DISCUSSION TITLE
Temporary files in edge function

### DISCUSSION THREAD

**Discussion by asyarifstudio** [supabase/supabase#21041] [Questions] [answered]
I plan to create a Youtube Video downloader. The server part will download the video and audio and merge them to temporary video file in the server and send it to the client. Can supabase edge function store temporary file like that? if yes, how big can it store? if no, any idea how to achieve this with supabase storage?

**Comment by GaryAustin1** [comment]
Supabase does not implement the deno file storage interface with edge functions.
https://supabase.com/docs/guides/functions/debugging#limitations

**Reply by asyarifstudio** [reply]
thanks @GaryAustin1 . do you know if it's possible to stream ffmpeg output to the storage?

**Reply by GaryAustin1** [reply]
Sorry I don't know. 
Info:
 https://supabase.com/docs/guides/storage
https://github.com/orgs/supabase/discussions/3773

**Reply by AntonOfTheWoods** [reply]
@GaryAustin1 the guide you refer to is now a 404 and the https://supabase.com/docs/guides/functions/limits page makes no mention at all of any FS limits. The "blocklist" still appears to be in place though...

**Reply by GaryAustin1** [reply, marked as answer]
There is now support for limited amount of temporary file space:  
https://supabase.com/docs/guides/functions/ephemeral-storage#limits
And there is support for static files:
https://github.com/orgs/supabase/discussions/32815

**Reply by AntonOfTheWoods** [reply]
Hmmm... Not really. You're not really allowed to use anything useful, and far, far more useful methods are blocked than just the two they mention there. I guess I can always compile my own... But that then means I will never be able to run it on their infra, which was one of the major attractions for developing for Supabase. Sigh.
